components/modal/shared.tsx

```diff
@@ -42,23 +45,58 @@ export const Footer: React.FC<
     onCancel,
     okButtonProps,
     cancelButtonProps,
+    footer,
   } = props;
 
   const [locale] = useLocale('Modal', getConfirmLocale());
 
+  // ================== Locale Text ==================
+  const okTextLocale = okText || locale?.okText;
+  const cancelTextLocale = cancelText || locale?.cancelText;
+
+  // ================= Context Value =================
+  const confirmBtnCtxValue: NormalOkBtnProps = {
+    confirmLoading,
+    okButtonProps,
+    okTextLocale,
+    okType,
+    onOk,
+  };
+
+  const cancelBtnCtxValue: NormalCancelBtnProps = {
+    cancelButtonProps,
+    cancelTextLocale,
+    onCancel,
+  };
+
+  const confirmBtnCtxValueMemo = React.useMemo(
+    () => confirmBtnCtxValue,
+    [...Object.values(confirmBtnCtxValue)],
+  );
+  const cancelBtnCtxValueMemo = React.useMemo(
+    () => cancelBtnCtxValue,
+    [...Object.values(cancelBtnCtxValue)],
+  );
+
+  const footerOriginNode = (
+    <>
+      <NormalCancelBtn />
+      <NormalOkBtn />
+    </>
+  );
+
   return (
     <DisabledContextProvider disabled={false}>
-      <Button onClick={onCancel} {...cancelButtonProps}>
-        {cancelText || locale?.cancelText}
-      </Button>
-      <Button
-        {...convertLegacyProps(okType)}
-        loading={confirmLoading}
-        onClick={onOk}
-        {...okButtonProps}
-      >
-        {okText || locale?.okText}
-      </Button>
+      <NormalOkBtnContextProvider value={confirmBtnCtxValueMemo}>
+        <NormalCancelBtnContextProvider value={cancelBtnCtxValueMemo}>
+          {typeof footer === 'function'
+            ? footer?.(footerOriginNode, {
```

Comment:
前面已经 typeof 了这里应该不需要判断了
```suggestion
            ? footer(footerOriginNode, {
```